File: app/src/main/java/com/raeed/tictactoc/UserInfoPage.java
package com.raeed.tictactoc;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.storage.FileDownloadTask;
import com.google.firebase.storage.FirebaseStorage;
import com.google.firebase.storage.StorageReference;
import com.google.firebase.storage.UploadTask;

import static com.raeed.tictactoc.SelectActivity.MAIN_PLAYER;
import static com.raeed.tictactoc.SelectActivity.MAIN_PLAYER_NAME;
import static com.raeed.tictactoc.SelectActivity.NOT_YET;
import static com.raeed.tictactoc.SelectActivity.telephone;


public class UserInfoPage extends AppCompatActivity {
    TextView name,phone;
    Button button_ok;
    ImageView image;
    Uri imageUri;
    Uri uri;
    FileDownloadTask FDT;

    SharedPreferences sharedPreferences;
    private FirebaseStorage storage;
    private StorageReference storageReference;

    private String Image;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_info_page);
        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
        MAIN_PLAYER_NAME = sharedPreferences.getString(MAIN_PLAYER, NOT_YET);
        telephone = sharedPreferences.getString("TelePhone", "0500000000");
        Image= sharedPreferences.getString("imagekey", "8f5eb8ce-1374-4cab-8722-4ccbdb54d6d9");
        MAIN_PLAYER_NAME = sharedPreferences.getString(MAIN_PLAYER, NOT_YET);
        storage = FirebaseStorage.getInstance();
        storageReference = storage.getReference();
        StorageReference riversRef = storageReference.child(Image);
        //String urlImage = storageReference.child("images/"+ Image).getPath();
        //String urlImage = "https://firebasestorage.googleapis.com/v0/b/tictactoc-8bbfd.appspot.com/o/images%2Fe3bc1266-627c-4fa9-83bf-eeed66f47896?alt=media&token=7fb3612b-242d-4074-a683-73c1efd3e3ff";
        name=findViewById(R.id.info_personname);
        name.setText(MAIN_PLAYER_NAME);
        phone=findViewById(R.id.info_telphone);
        phone.setText(telephone);
        image = findViewById(R.id.info_image);
        button_ok = findViewById(R.id.info_Ok);
        //Glide.with(UserInfoPage.this).load(urlImage).into(image);
        riversRef.getFile(uri);
        image.setImageURI(uri);
    }

    public void infodone(View view) {
        finish();
    }


}

File: app/src/main/java/com/raeed/tictactoc/UserInfoPage.kt
package com.raeed.tictactoc

import androidx.appcompat.app.AppCompatActivity
import android.widget.TextView
import android.content.SharedPreferences
import com.google.firebase.storage.FirebaseStorage
import com.google.firebase.storage.StorageReference
import android.graphics.Bitmap
import android.os.Bundle
import com.raeed.tictactoc.R
import android.preference.PreferenceManager
import com.raeed.tictactoc.SelectActivity
import com.google.android.gms.tasks.OnSuccessListener
import android.graphics.BitmapFactory
import android.view.View
import android.widget.Button
import android.widget.ImageView

class UserInfoPage : AppCompatActivity() {
      var name: TextView? = null
      var phone: TextView? = null
      var email: TextView? = null
      var button_ok: Button? = null
      var image: ImageView? = null
      lateinit var sharedPreferences: SharedPreferences
      private val storage = FirebaseStorage.getInstance("gs://tictactoc-8bbfd.appspot.com")
      private var profRf: StorageReference? = null
      private var storageReference: StorageReference? = null
      var bt: Bitmap? = null
      override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            setContentView(R.layout.activity_user_info_page)
            sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this)
            /*
                **************  Name section ********************
 */SelectActivity.MAIN_PLAYER_NAME =
                  sharedPreferences.getString(SelectActivity.MAIN_PLAYER, SelectActivity.NOT_YET)
            name = findViewById(R.id.info_personname)
            name?.setText(SelectActivity.MAIN_PLAYER_NAME)

            /*
         **************  Phone section ********************
         */SelectActivity.telephone =
                  sharedPreferences.getString(SelectActivity.PHONE_KEY, "0500000000")
            phone = findViewById(R.id.info_telphone)
            phone?.setText(SelectActivity.telephone)

            /*
         **************  Email section ********************
         */email = findViewById(R.id.info_email)
            email?.setText(sharedPreferences.getString(SelectActivity.EMAIL_KEY, "Email not Found"))

            /*
         **************  Image section ********************
         */image = findViewById(R.id.info_image)
            SelectActivity.IMAGE_NAME =
                  sharedPreferences.getString(SelectActivity.IMAGE_KEY, SelectActivity.IMAGE_NotYet)
            profRf = storage.reference.child(SelectActivity.PrfileImageChiled)
            storageReference = profRf!!.child(SelectActivity.IMAGE_NAME!!)
            storageReference!!.getBytes((1024 * 1024).toLong()).addOnSuccessListener { bytes ->
                  bt = BitmapFactory.decodeByteArray(bytes, 0, bytes.size)
                  image?.setImageBitmap(bt)
            }
            button_ok = findViewById(R.id.info_Ok)
      }

      fun infodone(view: View?) {
            finish()
      }
}